### ISSUE TITLE
[SPIR-V] ImageGather requires ImageGatherExtended capability

### ISSUE THREAD

**Issue by dirnerakos** [KhronosGroup/glslang#244]
For this fragment shader code

```
#version 310 es
#extension GL_EXT_gpu_shader5 : require

layout (location = 0) out mediump vec4 o_color;

layout (location = 0) in highp vec2 v_texCoord;

layout (binding = 0) uniform highp sampler2D u_sampler;
layout (binding = 1) uniform UOffset { highp ivec2 u_offset; };

void main(void)
{
	o_color = textureGatherOffset(u_sampler, v_texCoord, u_offset);
}
```

the generated SPIR-V code seems to be wrong, and the validation fails with the following error message:

```
Operand 6 of ImageGather requires one of these capabilities: ImageGatherExtended
```

**Comment by dirnerakos** [comment]
Similar error for shadow samplers:

```
Operand 6 of ImageDrefGather requires one of these capabilities: ImageGatherExtended
```

**Comment by johnkslang** [comment]
Fixed by pull request #277.
